scripts/viewer.js

```diff
@@ -0,0 +1,150 @@
+const BN = require("bn.js")
+const { getWithdrawableAmount, decodeOrders } = require("@gnosis.pm/dex-contracts")
+
+const { getExchange, getDeployedBrackets, fetchTokenInfoFromExchange } = require("./utils/trading_strategy_helpers")(
+  web3,
+  artifacts
+)
+const { default_yargs, checkBracketsForDuplicate } = require("./utils/default_yargs")
+const { fromErc20Units } = require("./utils/printing_tools")
+const { uniqueItems } = require("./utils/js_helpers")
+
+const argv = default_yargs
+  .option("masterSafe", {
+    type: "string",
+    describe: "address of Gnosis Safe whose brackets will be retrieved",
+  })
+  .option("brackets", {
+    type: "string",
+    describe: "comma-separated list of brackets to retrieve information about. Will be ignored if --masterSafe is provided",
+    coerce: (str) => {
+      return str.split(",")
+    },
+  })
+  .check(checkBracketsForDuplicate).argv
+
+const buildOptionalString = function ({ tokenData, requestedForWithdraw, storedInBracket }) {
+  const optionalStringAwaitingWithdraw = requestedForWithdraw.isZero()
+    ? ""
+    : `${fromErc20Units(requestedForWithdraw, tokenData.decimals)} awaiting to be withdrawn`
+  const optionalStringStoredInBracket = storedInBracket.isZero()
+    ? ""
+    : `${fromErc20Units(storedInBracket, tokenData.decimals)} directly stored in the bracket(s)`
+  const bothOptionalStrings = !requestedForWithdraw.isZero() && !storedInBracket.isZero()
+  const onlyOneOptionalStrings = !(requestedForWithdraw.isZero() == storedInBracket.isZero())
+  let fullOptionalString = ""
+  if (bothOptionalStrings) {
+    fullOptionalString = ` (of which ${optionalStringAwaitingWithdraw} and ${optionalStringStoredInBracket})`
+  } else if (onlyOneOptionalStrings) {
+    fullOptionalString = ` (of which ${optionalStringAwaitingWithdraw}${optionalStringStoredInBracket})`
+  }
+  return fullOptionalString
+}
+
+module.exports = async (callback) => {
+  try {
+    const exchangePromise = getExchange(web3)
+    let bracketAddresses
+    if (argv.masterSafe) {
+      bracketAddresses = await getDeployedBrackets(argv.masterSafe)
+      console.log(`Found ${bracketAddresses.length} brackets`)
+    } else {
+      if (!argv.brackets) {
+        callback("One of --masterSafe or --brackets=... is required.")
+      }
+      bracketAddresses = argv.brackets
+    }
```

Comment:
So if I provide both master safe and brackets arguments the brackets are ignored and `masterSafe` overrides it.